src/bundle/Resources/public/js/scripts/admin.notifications.filters.js

```diff
@@ -0,0 +1,107 @@
+        const checkboxes = filterNode.querySelectorAll(
+            '.ibexa-list-filters__item-content .ibexa-input--checkbox:not([name="dropdown-checkbox"])',
+        );
+        const timePicker = filterNode.querySelector('.ibexa-date-time-picker__input');
+
+        if (sourceSelect) {
+            sourceSelectOptions.forEach((option) => {
+                option.selected = false;
+            });
+
+            if (isNodeTimeFilter(filterNode)) {
+                sourceSelectOptions[0].selected = true;
+            }
+        } else if (checkboxes.length) {
+            checkboxes.forEach((checkbox) => (checkbox.checked = false));
+        } else if (timePicker.value.length) {
+            const formInput = filterNode.querySelector('.ibexa-picker__form-input');
+
+            timePicker.value = '';
+            formInput.value = '';
+
+            timePicker.dispatchEvent(new Event('input'));
+            formInput.dispatchEvent(new Event('input'));
+        }
+
+        searchForm.submit();
+    };
+    const attachFilterEvents = (filterNode) => {
+        if (!filterNode) {
+            return;
+        }
+
+        const sourceSelect = filterNode.querySelector('.ibexa-list-filters__item-content .ibexa-dropdown__source .ibexa-input--select');
+        const checkboxes = filterNode.querySelectorAll(
+            '.ibexa-list-filters__item-content .ibexa-input--checkbox:not([name="dropdown-checkbox"])',
+        );
+        const picker = filterNode.querySelector('.ibexa-input--date');
+
+        picker?.addEventListener('change', filterChange, false);
+        sourceSelect?.addEventListener('change', filterChange, false);
+        checkboxes.forEach((checkbox) => {
+            checkbox.addEventListener('change', filterChange, false);
+        });
+    };
+    const isNodeTimeFilter = (filterNode) => {
+        return filterNode.classList.contains('ibexa-picker');
+    };
+    const hasFilterValue = (filterNode) => {
+        if (!filterNode) {
+            return;
+        }
+
+        const select = filterNode.querySelector('.ibexa-dropdown__source .ibexa-input--select');
+        const checkedCheckboxes = filterNode.querySelectorAll('.ibexa-input--checkbox:checked');
+
+        if (isNodeTimeFilter(filterNode)) {
+            const timePicker = filterNode.querySelector('.ibexa-date-time-picker__input');
+
+            return !!timePicker.dataset.timestamp;
+        }
+
+        return !!(select?.value || checkedCheckboxes?.length);
+    };
+    const isSomeFilterSet = () => {
+        const hasStatusFilterValue = hasFilterValue(statusFilterNode);
+        const hasTypeFilterValue = hasFilterValue(typeFilterNode);
+        const hasDatetimeFilterValue = [...datetimeFilterNodes].some(hasFilterValue);
+
+        return hasStatusFilterValue || hasTypeFilterValue || hasDatetimeFilterValue;
+    };
+    const attachInitEvents = () => {
+        attachFilterEvents(statusFilterNode);
```

Comment:
Personally, I would prefer to have a different set of functions for each filter type instead of having one function with logic for every filter. Looking at `hasFilterValue` or `attachFilterEvents` it takes more time to understand what relates to what filter.